libsolidity/parsing/Parser.cpp

```diff
@@ -242,14 +245,21 @@ ASTPointer<ImportDirective> Parser::parseImportDirective()
 	return nodeFactory.createNode<ImportDirective>(path, unitAlias, move(symbolAliases));
 }
 
-ContractDefinition::ContractKind Parser::parseContractKind()
+ContractDefinition::ContractKind Parser::parseContractKind(bool *_isAbstract)
```

Comment:
Please use parameters only for input and return values for output. You can use an `std::tuple` or `std::pair`.